From 101657def81e8d1ff7522eedf9cec63a531ea2e6 Mon Sep 17 00:00:00 2001
From: Orlando Chamberlain <redecorating@protonmail.com>
Date: Thu, 10 Feb 2022 18:57:45 +1100
Subject: [PATCHv2 1/1] applesmc: Add iMacPro to applesmc_whitelist

The iMacPro1,1 is the only iMacPro released before the line was
discontinued. Add it to the applesmc_whitelist.

Signed-off-by: Orlando Chamberlain <redecorating@protonmail.com>
---
v1->v2: match "iMacPro" before matching "iMac" and "MacPro"

 drivers/hwmon/applesmc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/hwmon/applesmc.c b/drivers/hwmon/applesmc.c
index 8b3f73fcbb8a..b196f9939b28 100644
--- a/drivers/hwmon/applesmc.c
+++ b/drivers/hwmon/applesmc.c
@@ -1804,6 +1804,10 @@ static const struct dmi_system_id applesmc_whitelist[] __initconst = {
 	  DMI_MATCH(DMI_BOARD_VENDOR, "Apple"),
 	  DMI_MATCH(DMI_PRODUCT_NAME, "MacMini") },
 	},
+	{ applesmc_dmi_match, "Apple iMacPro", {
+	  DMI_MATCH(DMI_BOARD_VENDOR, "Apple"),
+	  DMI_MATCH(DMI_PRODUCT_NAME, "iMacPro") },
+	},
 	{ applesmc_dmi_match, "Apple MacPro", {
 	  DMI_MATCH(DMI_BOARD_VENDOR, "Apple"),
 	  DMI_MATCH(DMI_PRODUCT_NAME, "MacPro") },
-- 
2.34.1

